zerver/tests/test_channel_permissions.py

```diff
@@ -564,6 +564,7 @@ def check_unsubscribing_user(
             user: UserProfile,
             can_remove_subscribers_group: NamedUserGroup | UserGroupMembersData,
             expect_fail: bool = False,
+            user_has_channel_access: bool = True,
```

Comment:
Rather than having two parameters `expect_fail` and `user_has_channel_access`, this can have a `error_msg` parameter which will be `None` when user has permission.